src/recovery/tally.rs

```diff
@@ -18,9 +18,10 @@ use chain_impl_mockchain::{
     certificate,
     fragment::Fragment,
     ledger::{self, Ledger},
+    testing::scenario::FragmentFactory,
```

Comment:
Curious that we need to depend on things from the testing API, from the looks of it.

Reply:
Also, these use statements look messy, there's overlap with paths used above.